test-cloud/test-cloud-remote/src/test/java/org/kie/cloud/integrationtests/s2i/KieServerS2iDroolsIntegrationTest.java

```diff
@@ -0,0 +1,135 @@
+import org.kie.cloud.api.DeploymentScenarioBuilderFactory;
+import org.kie.cloud.api.scenario.GenericScenario;
+import org.kie.cloud.api.settings.DeploymentSettings;
+import org.kie.cloud.common.provider.KieServerClientProvider;
+import org.kie.cloud.integrationtests.AbstractCloudIntegrationTest;
+import org.kie.cloud.integrationtests.Kjar;
+import org.kie.cloud.maven.MavenDeployer;
+import org.kie.server.api.model.KieContainerResource;
+import org.kie.server.api.model.ReleaseId;
+import org.kie.server.api.model.ServiceResponse;
+import org.kie.server.client.KieServicesClient;
+import org.kie.server.client.RuleServicesClient;
+import org.kie.server.integrationtests.shared.KieServerReflections;
+
+public class KieServerS2iDroolsIntegrationTest extends AbstractCloudIntegrationTest<GenericScenario> {
+
+    private KieServicesClient kieServicesClient;
+    private RuleServicesClient ruleClient;
+
+    private static final String KIE_SESSION = "kbase1.stateless";
+    private static final String PERSON_CLASS_NAME = "org.kie.server.testing.Person";
+    private static final String PERSON_NAME = "Darth";
+    private static final String PERSON_SURNAME_FIELD = "surname";
+    private static final String PERSON_EXPECTED_SURNAME = "Vader";
+    private static final String PERSON_OUT_IDENTIFIER = "person1";
+
+    private static final Kjar DEPLOYED_KJAR = Kjar.STATELESS_SESSION;
+    private static final ReleaseId RELEASE_ID = new ReleaseId(DEPLOYED_KJAR.getGroupId(), DEPLOYED_KJAR.getName(), DEPLOYED_KJAR.getVersion());
+    private static final String KIE_CONTAINER_DEPLOYMENT = CONTAINER_ID + "=" + DEPLOYED_KJAR.toString();
+
+    private static final String REPO_BRANCH = "master";
+    private static final String PROJECT_SOURCE_FOLDER = "/kjars-sources";
+
+    private String repositoryName;
+    private ClassLoader kjarClassLoader;
+    private KieCommands commandsFactory = KieServices.Factory.get().getCommands();
+    private Set<Class<?>> extraClasses;
+
+    @Override
+    protected GenericScenario createDeploymentScenario(DeploymentScenarioBuilderFactory deploymentScenarioFactory) {
+        repositoryName = gitProvider.createGitRepositoryWithPrefix("KieServerS2iDroolsRepository", PROJECT_SOURCE_FOLDER);
+
+        DeploymentSettings kieServerS2Isettings = deploymentScenarioFactory.getKieServerS2ISettingsBuilder()
+                .withContainerDeployment(KIE_CONTAINER_DEPLOYMENT)
+                .withSourceLocation(gitProvider.getRepositoryUrl(repositoryName), REPO_BRANCH, DEPLOYED_KJAR.getName())
+                .withDroolsServerFilterClasses(false)
+                .build();
+
+        return deploymentScenarioFactory.getGenericScenarioBuilder()
+                .withKieServer(kieServerS2Isettings)
+                .build();
+    }
+
+    @Before
+    public void setUp() throws ClassNotFoundException {
+        deploymentScenario.getKieServerDeployments().get(0).setRouterTimeout(Duration.ofMinutes(5));
```

Comment:
In this test case should be rule processed immediately, right? So there is no need to set OpenShift router timeout.

Reply:
Right, it was a leftover, good catch, fixed.